docs/modders/Entities_Format/Town_Building_Format.md

```diff
@@ -248,3 +247,17 @@ In addition to above, it is possible to use same format as [Rewardable](../Map_O
 	"hidden" : false 
 }
 ```
+
+
+#### Markets in towns
+Market buildings require list of available [modes](../Map_Objects/Market.md)
+
+##### Marketplace
+```jsonc
+	"marketplace":    { "marketModes" : ["resource-resource", "resource-player"] },
+```
+
+##### Artifact merchant
+```jsonc
+	"special1":       { "type" : "artifactMerchant", "requires" : [ "marketplace" ], "marketModes" : ["resource-artifact", "artifact-resource"] },
+```
```

Comment:
Should also be mentioned in "Town building node" section (lines 113-175).

Reply:
Done. Looks like "type" : "artifactMerchant" is not needed. "marketModes" is enough.

Reply:
Somewhat.
Yeah, mechanics no longer need that type but I think it is still used for UI - to open correct mode when clicking on window. Same goes to other similar buildings - freelancers guild, marketplace, possibly - university of magic, etc.

Will take a look, already working on removing no longer necessary building types.